src/core/public/capabilities/capabilities_service.tsx

```diff
@@ -28,4 +33,22 @@ export interface UICapabilities {
   [key: string]: Record<string, boolean | Record<string, boolean>>;
 }
 
-export const uiCapabilities: UICapabilities = deepFreeze(chrome.getInjected('uiCapabilities'));
+/**
+ * Service that is responsible for UI Capabilities.
+ */
+export class CapabilitiesService {
+  public setup({ injectedMetadata }: StartDeps) {
+    return {
+      /**
+       * Get the UI capabilities.
+       */
+      getCapabilities: () =>
+        deepFreeze<UICapabilities>(injectedMetadata.getInjectedVar(
+          'uiCapabilities'
+        ) as UICapabilities),
+    };
+  }
+}
+
+/** @public */
+export type CapabilitiesSetup = ReturnType<CapabilitiesService['setup']>;
```

Comment:
would you mind to add an explicit contract? otherwise the documentation isn't descriptive enough https://github.com/elastic/kibana/pull/30585/files#diff-914fe2cfaec53bc4d487b248df58c10dR9

Reply:
Sure! Can you link to an example of how I might do this? I tried to find a core service that looked different from what I have, but I must be overlooking something here.

Reply:
we don't have examples for plugins yet. But for some core services:
https://github.com/elastic/kibana/blob/6251465110d0b5839f1e37daa632bb3291f99f22/src/core/server/elasticsearch/elasticsearch_service.ts#L35-L44

https://github.com/elastic/kibana/blob/6251465110d0b5839f1e37daa632bb3291f99f22/src/core/server/elasticsearch/elasticsearch_service.ts#L55